src/components/date-range-picker/date-range-picker.spec.ts

```diff
@@ -0,0 +1,102 @@
+    template: `
+        <ux-date-range-picker
+            [(start)]="start"
+            [(end)]="end"
+            [showTime]="showTime"
+            [showTimezone]="showTimezone"
+            [showSeconds]="showSeconds"
+            [showMeridian]="showMeridian"
+            [showSpinners]="showSpinners"
+            [showNowBtn]="showNowBtn"
+            [startTimezone]="startTimezone"
+            [endTimezone]="endTimezone"
+            (startChange)="onRangeChange()"
+            (endChange)="onRangeChange()"
+            (keydown.escape)="popover.hide(); input.focus()"
+            (startTimezoneChange)="onTimezoneChange(true, $event)"
+            (endTimezoneChange)="onTimezoneChange(false, $event)">
+        </ux-date-range-picker>
+    `
+})
+export class DateRangePickerComponent {
+
+    onRangeChange(): void { }
+    endChange(): void { }
+
+    start: Date;
+    end: Date;
+    date: string;
+    invalid: boolean = false;
+    showTime: boolean = false;
+    showTimezone: boolean = false;
+    showSeconds: boolean = false;
+    showMeridian: boolean = true;
+    showSpinners: boolean = true;
+    showNowBtn: boolean = false;
+    startTimezone: any = { name: 'GMT', offset: 0 };
+    endTimezone: any = { name: 'GMT', offset: 0 };
+}
+
+describe('Date Range Picker', () => {
+    let component: DateRangePickerComponent;
+    let fixture: ComponentFixture<DateRangePickerComponent>;
+    let nativeElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DateRangePickerModule, IconModule, PopoverModule, FormsModule],
+            declarations: [DateRangePickerComponent],
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(DateRangePickerComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    });
+
+    it('should initialise correctly', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should update start date and call onRangeChange when start date is changed ', async () => {
+        spyOn(component, 'onRangeChange');
+        component.start = new Date('Tue Jan 07 2020 00:00:00 GMT+0000 (Greenwich Mean Time)');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(getDate().innerHTML).toBe(' 7 January 2020 ');
+        expect(component.onRangeChange).toHaveBeenCalled();
+    });
+
+    it('should update end date and call onRangeChange when end date is changed', async () => {
+        spyOn(component, 'onRangeChange');
+        component.end = new Date('Thu Jan 23 2020 23:59:59 GMT+0000 (Greenwich Mean Time)');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        expect(getDate().innerHTML).toBe(' 23 January 2020 ');
+        expect(component.onRangeChange).toHaveBeenCalled();
+    });
+
```

Comment:
Above tests would need updates to `spyOn` correct functions. Also might be better to call the tests something like should call startChange when the end date is changed and should call startChange when the start date is changed ?